plugins/woocommerce/tests/php/src/Internal/PushNotifications/PushNotificationsTest.php

```diff
@@ -0,0 +1,189 @@
+use Automattic\WooCommerce\Internal\Features\FeaturesController;
+use Automattic\WooCommerce\Internal\PushNotifications\PushNotifications;
+use Automattic\WooCommerce\Proxies\LegacyProxy;
+use PHPUnit\Framework\MockObject\MockObject;
+use WC_Unit_Test_Case;
+
+/**
+ * PushNotifications test.
+ *
+ * @covers \Automattic\WooCommerce\Internal\PushNotifications\PushNotifications
+ */
+class PushNotificationsTest extends WC_Unit_Test_Case {
+	/**
+	 * @var JetpackConnectionManager|MockObject
+	 */
+	private $jetpack_connection_manager_mock;
+
+	/**
+	 * @var FeaturesController|MockObject
+	 */
+	private $features_controller_mock;
+
+	/**
+	 * Set up the test case.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->set_up_features_controller_mock();
+	}
+
+	/**
+	 * Tear down the test case.
+	 */
+	public function tearDown(): void {
+		global $wp_rest_server;
+		$wp_rest_server = null;
+
+		$this->reset_container_replacements();
+		wc_get_container()->reset_all_resolved();
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @testdox Tests the functionality is disabled if the feature flag is
+	 * disabled.
+	 */
+	public function test_it_can_tell_push_notifications_should_not_be_enabled_if_feature_is_disabled() {
+		$this->set_up_features_controller_mock( false );
+		$this->set_up_jetpack_connection_manager_mock( array( 'is_connected' ) );
+
+		$this->jetpack_connection_manager_mock
+			->expects( $this->never() )
+			->method( 'is_connected' );
+
+		$push_notifications = new PushNotifications();
+
+		$this->assertFalse( $push_notifications->should_be_enabled() );
+	}
+
+	/**
+	 * @testdox Tests the functionality is enabled if feature flag is enabled
+	 * and Jetpack is connected.
+	 */
+	public function test_it_can_tell_push_notifications_should_be_enabled_if_jetpack_is_connected() {
+		$this->set_up_jetpack_connection_manager_mock( array( 'is_connected' ) );
+
+		$this->jetpack_connection_manager_mock
+			->expects( $this->once() )
+			->method( 'is_connected' )
+			->willReturn( true );
+
+		$push_notifications = new PushNotifications();
+
+		$this->assertTrue( $push_notifications->should_be_enabled() );
+	}
+
+	/**
+	 * @testdox Tests the functionality is disabled if Jetpack is not connected
+	 * even when feature flag is enabled.
+	 */
+	public function test_it_can_tell_push_notifications_should_not_be_enabled_if_jetpack_is_not_connected() {
+		$this->set_up_jetpack_connection_manager_mock( array( 'is_connected' ) );
+
+		$this->jetpack_connection_manager_mock
+			->expects( $this->once() )
+			->method( 'is_connected' )
+			->willReturn( false );
+
+		$push_notifications = new PushNotifications();
+
+		$this->assertFalse( $push_notifications->should_be_enabled() );
+	}
+
+	/**
+	 * @testdox Tests that enablement state is cached within an instance.
+	 */
+	public function test_it_caches_enablement_state_correctly() {
+		/**
+		 * First instance with Jetpack disconnected - should return false.
```

Comment:
For single-line comments we often use the `// comment` syntax instead, even if a blank line is leave between the comment and the code to preserve readability.